Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs

```diff
@@ -9,6 +17,264 @@ namespace Unity.Multiplayer.Samples.BossRoom.Client
+        private DIScope _container;
+
+
+        [SerializeField] private LobbyUIManager m_lobbyUIManager;
+
+        [SerializeField] private CanvasGroup m_mainMenuButtons;
+        [SerializeField] private GameObject m_signInSpinner;
+
+        private void Awake()
+        {
+            m_mainMenuButtons.interactable = false;
+
+            CreateDIScope();
+        }
+
+        private void CreateDIScope()
+        {
+            void OnAuthSignIn()
+            {
+                Debug.Log("Signed in.");
+
+                m_mainMenuButtons.interactable = true;
+                m_signInSpinner.SetActive(false);
+
+                var localUser = _container.Resolve<LobbyUser>();
+                var identity = _container.Resolve<Identity>();
+                var localLobby = _container.Resolve<LocalLobby>();
+
+                localUser.ID = identity.GetSubIdentity(IIdentityType.Auth).GetContent("id");
+                // The local LobbyUser object will be hooked into UI before the LocalLobby is populated during lobby join, so the LocalLobby must know about it already when that happens.
+                localLobby.AddPlayer(localUser);
+            }
+
+            _container = new DIScope(DIScope.RootScope);
+
+            _container.BindMessageChannel<ClientUserSeekingDisapproval>();
+            _container.BindMessageChannel<DisplayErrorPopup>();
+
+            //todo: remember to cleanup unused message channels
+
+            _container.BindMessageChannel<RenameRequest>();
+            _container.BindMessageChannel<ClientUserApproved>();
+            _container.BindMessageChannel<UserStatus>();
+            _container.BindMessageChannel<StartCountdown>();
+            _container.BindMessageChannel<CancelCountdown>();
+            _container.BindMessageChannel<CompleteCountdown>();
+            _container.BindMessageChannel<ChangeGameState>();
+            _container.BindMessageChannel<ConfirmInGameState>();
+
+            _container.BindMessageChannel<EndGame>();
+
+            _container.BindAsSingle<LobbyAPIInterface>();
+            _container.BindAsSingle<LobbyAsyncRequests>();
+            _container.BindAsSingle<LocalGameState>();
+            _container.BindAsSingle<LobbyUser>();
+            _container.BindAsSingle<LobbyServiceData>();
+            _container.BindAsSingle<LobbyContentHeartbeat>();
+            _container.BindAsSingle<LocalLobby>();
+            _container.BindAsSingle<UIFactory>();
+
+            //var playerNetworkObject = Netcode.NetworkManager.Singleton.SpawnManager.GetPlayerNetworkObject(Netcode.NetworkManager.Singleton.LocalClientId);
+           // var persistentPlayer = playerNetworkObject.GetComponent<PersistentPlayer>();
+           // _container.BindInstanceAsSingle(persistentPlayer);
+
+            _container.BindInstanceAsSingle(m_lobbyUIManager);
+            _container.BindInstanceAsSingle(new Identity(OnAuthSignIn));
+
+            _container.FinalizeScopeConstruction();
```

Comment:
I get an error originating from here, when attempting to run the project from the Startup scene.
The Lobby UI is also visible by default, alongside the root Menu UI

```
NoInstanceToInjectException: Injection of type Unity.Multiplayer.Samples.BossRoom.PersistentPlayer failed.
BossRoom.Scripts.Shared.Infrastructure.DIScope.Resolve[T] () (at Assets/BossRoom/Scripts/Shared/Infrastructure/DIScope.cs:287)
BossRoom.Scripts.Shared.Infrastructure.DIScope.Resolve[T] () (at Assets/BossRoom/Scripts/Shared/Infrastructure/DIScope.cs:284)
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <695d1cc93cca45069c528c15c9fdd749>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <695d1cc93cca45069c528c15c9fdd749>:0)
BossRoom.Scripts.Shared.Infrastructure.DIScope.Inject (System.Object obj) (at Assets/BossRoom/Scripts/Shared/Infrastructure/DIScope.cs:327)
BossRoom.Scripts.Shared.Infrastructure.DIScope.FinalizeScopeConstruction () (at Assets/BossRoom/Scripts/Shared/Infrastructure/DIScope.cs:310)
Unity.Multiplayer.Samples.BossRoom.Client.ClientMainMenuState.CreateDIScope () (at Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs:97)
Unity.Multiplayer.Samples.BossRoom.Client.ClientMainMenuState.Awake () (at Assets/BossRoom/Scripts/Client/Game/State/ClientMainMenuState.cs:42)
```

Reply:
git user error related